File: range-sum-of-bst/range-sum-of-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        
        result = 0
        
        queue = deque()
        queue.append(root)
        
        while queue:
            node = queue.popleft()
            if node.val >= low and node.val <= high:
                result += node.val
                if node.left is not None:
                    queue.append(node.left)
                if node.right is not None:
                    queue.append(node.right)
            elif node.val < low and node.right is not None:
                queue.append(node.right)
            elif node.val > high and node.left is not None:
                queue.append(node.left)

        return result

File: range-sum-of-bst/range-sum-of-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        Queue<TreeNode> queue = new LinkedList<>();
        int result = 0;
        
        queue.offer(root);
        
        while(!queue.isEmpty()) {
            TreeNode node = queue.poll();
            if(node.val >= low && node.val <= high) {
                result += node.val;
                if(node.left != null)
                    queue.offer(node.left);
                if(node.right != null)
                    queue.offer(node.right);
            } else if(node.val < low) {
                if(node.right != null)
                    queue.offer(node.right);
            } else if(node.val > high) {
                if(node.left != null)
                    queue.offer(node.left);
            }
        }
        return result;
    }
}